src/sqs/mod.rs

```diff
@@ -0,0 +1,313 @@
+use std::rc::Rc;
+use std::time::Duration;
+use std::convert::TryFrom;
+
+use rocket::Config;
+use rusoto_core::Region;
+use rusoto_sqs::{DeleteMessageRequest, Message, ReceiveMessageRequest, Sqs, SqsClient};
+use serde_json;
+
+use error::Result;
+use failure::{self, err_msg};
+use logging::RBLogger;
+
+#[derive(Default, Debug, Clone)]
+pub struct SyncEvent {
+    pub event: String, // act on "delete", "device:delete"; ignore others
+    pub uid: String,
+    pub id: String,
+    pub handle: String,
+}
+
+impl TryFrom<Message> for SyncEvent {
+    type Error = failure::Error;
+
+    fn try_from(msg: Message) -> Result<Self> {
+        let body = match msg.body {
+            Some(d) => d,
+            None => {
+                return Err(err_msg("Missing body"))
+            }
+        };
+        let parsed: serde_json::Value = serde_json::from_str(&body).unwrap();
+        let blank = json!("");
+        let device = parsed.get("id").unwrap_or(&blank);
+        let handle = msg.receipt_handle.unwrap_or("".to_owned());
+        Ok(Self {
+            event: parsed["event"].as_str().unwrap_or("").to_owned(),
+            uid: parsed["uid"].as_str().unwrap_or("").to_owned(),
+            id: device.as_str().unwrap_or("").to_owned(),
+            handle,
+        })
+    }
+}
+
+#[derive(Clone)]
+pub struct SyncEventQueue {
+    sqs: Rc<SqsClient>,
+    url: String,
+    logger: RBLogger,
+}
+
+impl SyncEventQueue {
+    pub fn from_config(config: &Config, logger: &RBLogger) -> SyncEventQueue {
+        let region = env::var("AWS_LOCAL_SQS")
+            .map(|endpoint| Region::Custom {
+                endpoint,
+                name: "env_var".to_string(),
+            })
+            .unwrap_or_default();
+        // "https://sqs.us-east-1.amazonaws.com/927034868273/fxa-oauth-account-change-dev"
+        let queue_url = config
+            .get_str("sqs_url")
+            .unwrap_or("https://sqs.us-east-1.amazonaws.com/927034868273/rust_sqs_test");
+        // Max wait_time is 20, so we'll pound away on SQS.
+        SyncEventQueue {
+            sqs: Rc::new(SqsClient::simple(region)),
+            url: queue_url.to_owned(),
+            logger: logger.clone(),
+        }
+    }
+
+    //to "ack" an SQS message, you delete it from the queue.
+    pub fn ack_message(&self, event: &SyncEvent) -> Result<()> {
+        let sqs = self.sqs.clone();
+        let mut request = DeleteMessageRequest::default();
+        request.queue_url = self.url.clone();
+        request.receipt_handle = event.handle.clone();
+        sqs
+            .delete_message(&request)
+            .with_timeout(Duration::new(1, 0))
+            .sync()?;
+        Ok(())
+    }
+
+    // Fetch message from sqs
+    pub fn fetch(&self) -> Option<SyncEvent> {
+        let sqs = self.sqs.clone();
+        // capture response via retry_if(move||{..})
+        let mut request = ReceiveMessageRequest::default();
+        request.queue_url = self.url.clone();
+        match sqs
```

Comment:
To avoid 2 layers of indentation below, perhaps:
```rust
let response = match sqs.receive_message(&request).with_timeout(Duration::new(1, 0)).sync() {
    Ok(r) => r,
    Err(e) => { ..., return None }
};
// Proceed with response object
```

Reply:
Perhaps. I was trying to limit the number of indented `return`s I'll muck about and see if I can come up with something better.